File: 1.1/src/cz/mg/vulkan/PFNvkDestroySampler.java
package cz.mg.vulkan;

public class PFNvkDestroySampler extends VkFunctionPointer {
    public PFNvkDestroySampler() {
    }

    protected PFNvkDestroySampler(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroySampler(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroySampler(long value) {
        setValue(value);
    }

    public PFNvkDestroySampler(VkInstance instance) {
        super(instance, new VkString("vkDestroySampler"));
    }

    public void call(VkDevice device, VkSampler sampler, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), sampler != null ? sampler.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long sampler, long pAllocator);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkDestroySampler.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroySampler_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong sampler, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroySampler f = (PFN_vkDestroySampler)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSampler*)jniLongToPointer(sampler)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

